plugin/pkg/scheduler/algorithm/predicates/metadata_test.go

```diff
@@ -0,0 +1,363 @@
+/*
+Copyright 2017 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package predicates
+
+import (
+	"fmt"
+	"reflect"
+	"sort"
+	"testing"
+
+	"k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/kubernetes/plugin/pkg/scheduler/schedulercache"
+	schedulertesting "k8s.io/kubernetes/plugin/pkg/scheduler/testing"
+)
+
+// sortableAntiAffinityTerms lets us to sort anti-affinity terms.
+type sortableAntiAffinityTerms struct {
+	Items []matchingPodAntiAffinityTerm
+}
+
+// Less establishes some ordering between two matchingPodAntiAffinityTerms for
+// sorting.
+func (s *sortableAntiAffinityTerms) Less(i, j int) bool {
+	t1, t2 := s.Items[i], s.Items[j]
+	return t1.node.Name < t2.node.Name ||
+			(t1.node.Name == t2.node.Name && len(t1.term.Namespaces) < len(t2.term.Namespaces)) ||
```

Comment:
This is a bit unfortunate.
To make it a bit more readable, we can try changing to:
```
if t1.node.Name != t2.node.Name {
  return t1.node.Name < t2.node.Name
}
// and now compare the rest similarly
```

It's effectively the same, but simpler to follow.